File: security_assistant/remediation/code_examples/sql_injection_fix.go
// VULNERABLE:
// query := "SELECT * FROM users WHERE username = '" + username + "'"
// rows, err := db.Query(query)

// SECURE (Parameterized Query):
query := "SELECT * FROM users WHERE username = ?"
rows, err := db.Query(query, username)
if err != nil {
    log.Fatal(err)
}
defer rows.Close()

File: security_assistant/remediation/code_examples/sql_injection_fix.rs
// VULNERABLE:
// let query = format!("SELECT * FROM users WHERE username = '{}'", username);
// let rows = conn.query(&query, &[])?;

// SECURE (Parameterized Query with rusqlite):
let query = "SELECT * FROM users WHERE username = ?1";
let mut stmt = conn.prepare(query)?;
let rows = stmt.query(params![username])?;
